js/src/Ice/InputStream.js

```diff
@@ -928,98 +910,75 @@ class PatchEntry {
 
 export class InputStream {
     constructor(arg1, arg2, arg3) {
-        const args = {
-            instance: null,
-            encoding: null,
-            bytes: null,
-            buffer: null,
-        };
-        this._checkArgs([arg1, arg2, arg3], args);
-        this._initialize(args);
-    }
-
-    _checkArgs(arr, args) {
-        //
-        // The constructor can accept a variety of argument combinations:
-        //
-        // ()
-        // (communicator)
-        // (instance)
-        // (encoding)
-        // (array)
-        // (buffer)
-        // (communicator, encoding)
-        // (instance, encoding)
-        // (communicator, array)
-        // (instance, array)
-        // (communicator, buffer)
-        // (instance, buffer)
-        // (communicator, encoding, array)
-        // (instance, encoding, array)
-        // (communicator, encoding, buffer)
-        // (instance, encoding, buffer)
-        // (encoding, array)
-        // (encoding, buffer)
-        //
-        arr.forEach(arg => {
-            if (arg !== null && arg !== undefined) {
-                if (arg.constructor === Communicator) {
-                    args.instance = arg.instance;
-                } else if (arg.constructor === Instance) {
-                    args.instance = arg;
-                } else if (arg.constructor === EncodingVersion) {
-                    args.encoding = arg;
-                } else if (arg.constructor === Buffer) {
-                    args.buffer = arg;
-                } else if (arg.constructor === ArrayBuffer) {
-                    args.bytes = arg;
-                } else if (arg.constructor === Uint8Array) {
-                    args.bytes = arg.buffer;
+        // The first argument must be a Communicator or an Instance.
+        if (arg1 === null || arg1 === undefined) {
+            throw new InitializationException("missing argument(s) for InputStream constructor");
+        }
+
+        if (arg1.constructor === Instance) {
+            // (instance, encoding) or (instance, encoding, buffer)
+
+            this._instance = arg1;
+            if (arg2 === null || arg2 === undefined || arg2.constructor !== EncodingVersion) {
+                throw new InitializationException(
+                    "expected the encoding version as the second argument to the InputStream constructor",
+                );
+            }
+            this._encoding = arg2;
+
+            if (arg3 === undefined || arg3 === null) {
```

Comment:
I think it is a bit more readable to do:

```
if (arg3 === undefined || arg3 === null) {
    this._buf = new Buffer();
} else if (arg3.constructor === Buffer) {
    this._buf = arg3;
} else {
    throw new InitializationException("unknown argument to InputStream constructor");
}
```

Reply:
Fixed.